ambry-router/src/main/java/com.github.ambry.router/PutOperation.java

```diff
@@ -798,6 +799,15 @@ String getServiceId() {
     return passedInBlobProperties.getServiceId();
   }
 
+  /**
+   * This will return a view of the correlation IDs of requests that may still be in flight. This is not threadsafe
+   * and should only be called from the main event loop.
+   * @return the set of correlation IDs of requests that are still in flight.
+   */
+  Set<Integer> getInFlightCorrelationIds() {
+    return correlationIdToPutChunk.keySet();
```

Comment:
Are all reads/writes to `correlationIdToPutChunk` done by the main loop?  What other threads touch `PutOperation`?

Reply:
Yep this is a thread unsafe map to begin with and all operations are done from the poll and handleResponse methods, which are called in the main loop.